File: 226. Invert Binary Tree/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        dfs(root);
        return root;
    }
    public void dfs(TreeNode curr) {
        if (curr == null) return;
        TreeNode holder = curr.left;
        curr.left = curr.right;
        curr.right = holder;
        dfs(curr.left);
        dfs(curr.right);
    }
}

File: 226. Invert Binary Tree/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.dfs(root)
        return root
    def dfs(self, curr):
        if not curr: return
        curr.left, curr.right = curr.right, curr.left
        self.dfs(curr.left)
        self.dfs(curr.right)